src/cache.jl

```diff
@@ -95,16 +95,98 @@ end
 
 getindex(A::AbstractCachedMatrix, I::Integer) = A[Base._to_subscript_indices(A, I)...]
 
-@inline getindex(A::AbstractCachedMatrix, kr::AbstractUnitRange, jr::AbstractUnitRange) = layout_getindex(A, kr, jr)
-@inline getindex(A::AbstractCachedMatrix, kr::AbstractVector, jr::AbstractVector) = layout_getindex(A, kr, jr)
-@inline getindex(A::AbstractCachedMatrix, k::Integer, jr::AbstractVector) = layout_getindex(A, k, jr)
+@inline function getindex(A::AbstractCachedMatrix, k::Integer, jr::AbstractUnitRange)
```

Comment:
This definition won't work since its possible that `jr` is infinite. here's an option:
```suggestion
function _isfinite_getindex(A, kr, jr, maxkr::Int, maxjr::Int)
     resizedata!(A, maxkr, maxjr)
     getindex(A.data, kr, jr)
end
_isfinite_getindex(A, kr, jr, _, _) = layout_getindex(A, kr, jr)

@inline getindex(A::AbstractCachedMatrix, k::Integer, jr::AbstractUnitRange) = _isfinite_getindex(A::AbstractCachedMatrix, k, jr, maximum(k), maximum(jr))
# TODO other getindex
```